crypto/bio/bio_mem.c

```diff
@@ -65,10 +65,16 @@
 
 #include "../internal.h"
 
+typedef struct bio_buf_mem_st {
+  struct buf_mem_st *buf;   /* allocated buffer */
+  size_t read_off;          /* number of bytes already read from the buffer */
+} BIO_BUF_MEM;
+
 
 BIO *BIO_new_mem_buf(const void *buf, ossl_ssize_t len) {
```

Comment:
We cast `len` to a `size_t` below. Although unlikely, we should verify that `len <= SIZE_MAX` to avoid overflow.